packages/nextjs/test/integration/test/server.js

```diff
@@ -0,0 +1,88 @@
+const fs = require('fs').promises;
+const { createServer } = require('http');
+const { parse } = require('url');
+const path = require('path');
+
+const next = require('next');
+
+const { DEBUG_MODE } = require('./utils');
+const { log } = console;
+
+/**
+ * Usage: node server.js <filename>
+ *
+ * ENV Variables:
+ * DEBUG=[bool] - enable request and application logs
+ * DEBUG_DEPTH=[int] - set logging depth
+ *
+ * Arguments:
+ * <filename> - filter tests based on filename partial match
+ */
+
+const FILES_FILTER = process.argv[2];
+
+(async () => {
+  let scenarios = await fs.readdir(path.resolve(__dirname, './server'));
+
+  if (FILES_FILTER) {
+    scenarios = scenarios.filter(file => file.toLowerCase().includes(FILES_FILTER));
```

Comment:
Purely for convinience. It just annoyed me that I typed `node server.js api` and it didn't find the test ¯\_(ツ)_/¯ 
I dont understand the second part of your feedback. Line 19 states that it'll match on partials 🤔

Reply:
What I understand from "partial match" is it can be a substring, not that we're not considering the case. I'm good for not considering it, but I feel we should not that.